pkg/cmd/search/repos/repos.go

```diff
@@ -0,0 +1,238 @@
+package repos
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/pkg/cmd/search/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/export"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/pkg/search"
+	"github.com/cli/cli/v2/pkg/text"
+	"github.com/cli/cli/v2/utils"
+	"github.com/spf13/cobra"
+)
+
+type ReposOptions struct {
+	Browser      cmdutil.Browser
+	Config       func() (config.Config, error)
+	Exporter     cmdutil.Exporter
+	GoTemplate   string
+	HttpClient   func() (*http.Client, error)
+	IO           *iostreams.IOStreams
+	JqExpression string
+	Query        search.Query
+	WebMode      bool
+}
+
+func NewCmdRepos(f *cmdutil.Factory, runF func(*ReposOptions) error) *cobra.Command {
+	opts := &ReposOptions{
+		Browser:    f.Browser,
+		Config:     f.Config,
+		HttpClient: f.HttpClient,
+		IO:         f.IOStreams,
+		Query:      NewSearchQuery(),
+	}
+
+	cmd := &cobra.Command{
+		Use:   "repos [<query>]",
+		Short: "Search repositories",
+		RunE: func(c *cobra.Command, args []string) error {
+			opts.Query.Keywords = args
+			err := cmdutil.MutuallyExclusive("expected exactly one of `--jq`, `--template`, or `--web`",
+				opts.GoTemplate != "",
+				opts.JqExpression != "",
+				opts.WebMode)
+			if err != nil {
+				return err
+			}
+			if opts.Query.Limit < 1 || opts.Query.Limit > 1000 {
+				return cmdutil.FlagErrorf("`--limit` must be between 1 and 1000")
+			}
+			if runF != nil {
+				return runF(opts)
+			}
+			return reposRun(opts)
+		},
+	}
+
+	// Output flags
+	cmd.Flags().StringVarP(&opts.GoTemplate, "template", "t", "", "Format JSON output using a Go template")
+	cmd.Flags().StringVarP(&opts.JqExpression, "jq", "q", "", "Format JSON output using a jq `expression`")
```

Comment:
I would suggest still supporting a `--json` flag that explicitly specifies fields to fetch, even if you're currently using REST, in case we switch the underlying lookup to GraphQL.

And to avoid re-implementing the search functionality for issues, PRs, and repos, you could largely reuse the search functions we already have in place: https://github.com/cli/cli/blob/2c981b70bf653ca12f188171cefe4115c03c9c3a/pkg/cmd/issue/list/http.go#L115-L115